### PR TITLE
Enhanced RDoc for several methods

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods:
- each
- read
- header_row?
- shift
- inspect

### CODE REVIEW

File: lib/csv.rb [hunk 1]
```diff
@@ -2260,25 +2260,56 @@ def header_convert(name = nil, &converter)
 
   include Enumerable
 
+  # :call-seq:
+  #   csv.each -> self
```
Comment (kou) [inline lib/csv.rb:2264]: Could you change the return value to "undefined"?

```suggestion
  #   csv.each -> enumerator
  #   csv.each {|row| ...}
```

File: lib/csv.rb [hunk 2]
```diff
@@ -2260,25 +2260,56 @@ def header_convert(name = nil, &converter)
...
+  #   csv.each do |row|
+  #     p row
+  #   end
   def each(&block)
     parser_enumerator.each(&block)
   end
 
   # :call-seq:
-  #     read
+  #   csv.read -> array or csv_table
```
Comment (kou) [inline lib/csv.rb:2306]: How about splitting each pattern?

```suggestion
  #   csv.read -> array
  #   csv.read -> csv_table (CSV::Table is better?)
```

Comment (BurdetteLamar) [reply lib/csv.rb:2307]: > What I have here is what I see in the doc for such as Hash and Array.  People have been careful to use something (csv_table) that is English, not code.  I once wrote foo.to_h -> Hash, and had someone point out that it's somewhat ambiguous:  class Hash or instance of Hash?

Comment (kou) [reply lib/csv.rb:2307]: I see.

**Pull request merged**